dnf/base.py

```diff
@@ -1744,9 +1753,111 @@ def install_module(self, specs, strict=True):
         """
         return self.repo_module_dict.install(specs, strict)
 
+    def classify(self, exclude, install):
+        install_specs = argparse.Namespace()
+        exclude_specs = argparse.Namespace()
+        classify_specs(install_specs, install)
+        classify_specs(exclude_specs, exclude)
+
+        return exclude_specs, install_specs
+
+    def exclude_spec(self, exclude_specs):
+        glob_excludes = [exclude for exclude in exclude_specs.pkg_specs
+                         if dnf.util.is_glob_pattern(exclude)]
+        excludes = [exclude for exclude in exclude_specs.pkg_specs
+                    if exclude not in glob_excludes]
+        exclude_query = self.sack.query().filter(name=excludes)
+        glob_exclude_query = self.sack.query().filter(name__glob=glob_excludes)
+        self.sack.add_excludes(exclude_query)
+        self.sack.add_excludes(glob_exclude_query)
+
+    def install_packages(self, package_specs, ignored):
+        for spec in package_specs:
+            try:
+                self.install(spec, strict=self.conf.strict)
```

Comment:
Don't take strict setting from dnf.conf directly. Better to inherit it. additionally reponame and forms could be inherited.